### PR TITLE
Add 5 proposer test

### PR DESCRIPTION
**Pull request by cmr**
This makes the multiproposer test take a `-num-proposers`, and run for 3 minutes. It also rewrites the prefix check to be tolerant of the many forks that happen during Ouroboros consensus. It does this by querying the "best path" (ie, list of state hashes from the tf root to tf best tip) for each node and ensuring they agree in _some_ node in the tree. If ever a path becomes disjoint, the node has forked and the test fails.

### CODE REVIEW

File: src/app/cli/src/coda_shared_prefix_multiproposer_test.ml [hunk 1]
```diff
@@ -5,18 +5,21 @@ open Coda_main
...
+let main n () =
   let logger = Logger.create () in
-  let n = 2 in
   let snark_work_public_keys i = None in
   let%bind testnet =
     Coda_worker_testnet.test logger n Option.some snark_work_public_keys
       Protocols.Coda_pow.Work_selection.Seq
   in
-  let%bind () = after (Time.Span.of_sec 30.) in
+  let%bind () = after (Time.Span.of_min 10.) in
```
Comment (johnwu93) [inline src/app/cli/src/coda_shared_prefix_multiproposer_test.ml]: In 10 minutes, what are the things we are expecting the test to finish? 
Like how many blocks are we expecting each proposer to have?

Comment (cmr) [reply src/app/cli/src/coda_shared_prefix_multiproposer_test.ml]: Concretely we want to be able to have this test have caught the "empty epoch bug", which means it needs to run for at least 2 full epochs after the first ledger proof is emitted. I haven't done all the math to figure how long that is yet -- `10m` is long enough that I usually see the flakes cause it to fail. anything much shorter tends to sometimes hide the errors.

**Comment by cmr** [comment]
seems the new shared prefix check isn't quite as stable -- shame. I'll investigate why...

**Comment by cmr** [comment]
it was less stable because of https://github.com/CodaProtocol/coda/pull/2011. This should be good to go now.

**Comment by cmr** [comment]
latest failures are `best_path` RPC getting interrupted when we restart the node.

**Comment by cmr** [comment]
These new commits need review:

https://github.com/CodaProtocol/coda/pull/1870/commits/00c63b99e97879444b421a8d2d2991e63373306e

https://github.com/CodaProtocol/coda/pull/1870/commits/8dd0552d7e8665fe5728af5f31dea076e3b227c4

https://github.com/CodaProtocol/coda/pull/1870/commits/1a071d34af0ba7fc6a3ca2cbaaaebc4bfa66bc01

**Comment by bkase** [comment]
Nits on 00c63b9 but looks good to me!

**Pull request merged**